nodestream/project/pipeline_scope.py

```diff
@@ -26,26 +26,34 @@ def __init__(
         pipelines: List[PipelineDefinition],
         persist: bool = True,
         config: ScopeConfig = None,
-        targets: List[str] = None,
+        targets: List[str] = [],
```

Comment:
I am thinking about the name `targets`.

To me I would expect `targets` to be a `List[Target]` not a `List[str]`. Three possible options:

- I get over it 
- Change `targets` to `target_names` 
- Resolve the actual names into `Target` objects and reference the `Target`s instead. 

I am very slightly leaning towards the third option, but it is more work. Thoughts?

Reply:
We will change to Target types but agreed on this being part of a larger refactor since this isn't easily doable at the moment.